src/oop/static_modifier/hard/q6/MainTest.java

```diff
@@ -0,0 +1,217 @@
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"), String.format(wrongModifierMessage, staticAttributeName, className));
+        assertTrue(testClass.hasField(attributeName2), String.format(missingAttributeMessage, className, attributeName2));
+        assertTrue(testClass.hasField(attributeName2, int.class), String.format(wrongTypeMessage, attributeName2, className));
+        assertTrue(testClass.hasModifier(attributeName2, "private"), String.format(wrongModifierMessage, attributeName2, className));
+    }
+
+    @Test
+    @Order(3)
+    public void farmHasRequiredConstructor() {
+        String missingConstructorMessage = "The %s class is missing a required constructor. Make sure that it is named correctly and has the correct parameters.";
+        String wrongAccessModifier = "The %s class constructor has the wrong visibility modifier. Make sure that it is visible from all other classes.";
+        Class<?>[] constructorArgs = new Class[]{double.class, int.class};
+        assertTrue(testClass.hasConstructor(constructorArgs), String.format(missingConstructorMessage, className));
+        assertTrue(testClass.hasModifier(constructorArgs, "public"), String.format(wrongAccessModifier, className));
+    }
+
+    @Test
+    @Order(1)
+    @Tag("dependency1")
+    public void staticVariableIsInitializedCorrectly() {
+        _assertEquals(initialBeams, testClass.getFieldValue(null, staticAttributeName),
+                "Your " + staticAttributeName + " static attribute is not initialized correctly.");
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(2.5, 3, 6.0),
+                Arguments.of(1.7, 5, 10.0),
+                Arguments.of(3.1, 4, 15.0),
+                Arguments.of(2.3, 2, 4.0),
+                Arguments.of(1.9, 6, 10.5),
+                Arguments.of(2.8, 5, 12.0),
+                Arguments.of(3.4, 3, 11.0),
+                Arguments.of(1.6, 4, 7.5),
+                Arguments.of(2.5, 7, 18.0),
+                Arguments.of(3.0, 3, 9.5)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    @Tag("dependency1")
+    @Order(4)
+    public void farmConstructorInitializesValuesCorrectly(double value1, int value2) throws Throwable {
+        String wrongValueMessage = "The %s constructor did not initialize the %s attribute to the correct value based on the parameters passed to the constructor.";
+        Object[][] constructorArgs = {
+                {value1, double.class},
+                {value2, int.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName1), String.format(wrongValueMessage, className, attributeName1));
+        _assertEquals(value2, testClass.getFieldValue(checkupInstance, attributeName2), String.format(wrongValueMessage, className, attributeName2));
+    }
```

Comment:
add a test that checks they don't alter the static value on calling the constructor